specifyweb/specify/views.py

```diff
@@ -501,36 +509,55 @@ def update_record(record: models.Model):
                 if response is not None and response.status_code != 204:
                     return response
 
-    # Dedupe by deleting the agent that is being replaced and updating the old model ID to the new one
-    target_model.objects.get(id=old_model_id).delete()
-
+    # Dedupe by deleting the record that is being replaced and updating the old model ID to the new one
+    for old_model_id in old_model_ids:
+        target_model.objects.get(id=old_model_id).delete()
+
+    # Update new record with json info, if given
+    has_new_record_info = False if new_record_info is None else True
```

Comment:
can be simplified to `has_new_record_info = new_record_info is not None`

Reply:
Fixed in 2b10695